static/js/operation.js

```diff
@@ -165,89 +127,69 @@ function parseAssembly(text, data) {
-  var contig_length = 0;
-  var GC_count = 0;
+  var currentContigLength = 0;
+  var gcCount = 0;
 
   for (var i = 0; i < lines.length; i++) {
     var line = lines[i];
-    if (line.toUpperCase().match(/^[ATGCU]+$/g)) { // Checking if the current line is not a contig title
-      [GC_count, contig_length] = calculateGCAndLength(line, GC_count, contig_length);
-      gc = ((GC_count/contig_length).toFixed(2)).toString(); // idnetified last component of contig title
+    if (line.search(/^>{1}/gm) === 0) { // Checking if the current line is a contig title
+        [id, length, coverage] = parseContigTitles(line, format);
+        if (length <= minContigLength) 
+          continue;
+
+        // append dataframe with contig title information
+        df.push([id, length, , coverage]) // gc not calculated yet
+        gcCount = 0;
+        currentContigLength = 0;
     }
     else {
-      if (i !== 0) {
-        if (length <= 1000) {
-          continue;
-        }
-        df.push([id, length, gc, coverage]) // append dataframe with contig title information
-        GC_count = 0;
-        contig_length = 0;
+      gcCount += calculateLineGC(line);
+      currentContigLength += line.length; 
+      if (df.length != 0 && currentContigLength === Number(length)) { // verifying if current line is last line in contig
+        gc = (gcCount/currentContigLength).toString(); // identified last component of contig title
+        df[df.length - 1][2] = gc; // update gc in dataframe
       }
-      [id, length, coverage] = parseContigTitles(line, format);
-      coverage = (Number(coverage).toFixed(3)).toString();
     }
   }
-  if (df.length === 0) throw 'Error: No contig is bigger than 1000 base pairs.';
-
 
-  var deci = {};
-  var cats = {};
-  var feats = {};
-
-  // identify field types and re-format data
-  var types = ['id'];
-  for (var i = 1; i < cols.length; i++) {
-    var arr = [];
-    for (var j = 0; j < df.length; j++) {
-      arr.push(df[df.length - j - 1][i]);
-    }
-      var x = parseFieldType(cols[i], arr);
-      var type = x[0];
-      var col = x[1];
-      types.push(type); // identified type
-      cols[i] = col; // updated name
-      for (var j = 0; j < df.length; j++) {
-        df[j][i] = arr[j];
-      }
-  }
-    // summarize categories or features
-    switch (type) {
-      case 'number':
-        deci[col] = maxDecimals(arr);
-        break;
-      case 'category':
-        cats[col] = listCats(arr);
-        break;
-      case 'feature':
-        feats[col] = listFeats(arr);
-        break;
-    }
+  if (df.length === 0) throw 'Error: No contig is bigger than ' + minContigLength + ' base pairs.';
 
-  data.cols = cols;
-  data.types = types;
-  data.features = [];
-  data.df = df;
-  return [deci, cats, feats];
+  return formatData(data, df, cols);
 }
 
 /**
- * Calculates GC count in and length of one line in the contig.
- * @function calculateGCAndLength
+ * Calculates GC count in one line in the contig.
```

Comment:
Can add a link to the IUPAC webpage so that future developers know where you got those values:
```
* {@link https://www.bioinformatics.org/sms/iupac.html}
```

See [here](https://jsdoc.app/tags-inline-link.html) for the JavaScript docstring syntax.